### PR TITLE
Contact Attributes

### LINKED ISSUE #215: Rename set_contact_property.field_name to property
To match https://github.com/nyaruka/goflow/pull/200

### LINKED ISSUE #221: Contact field logic booched
updating a field downs the app because we aren't giving the new action a type; we need to update the logic in `FieldElement` ([here](https://github.com/nyaruka/floweditor/blob/master/src/component/actions/SetContactField/SetContactFieldForm.tsx#L59)) to ensure the right type is added now that we support 2 types.

### PR DESCRIPTION
**Pull request by ycleptkellan**
fixes #221, resolves #215 

- `set_contact_property.field_name` => `set_contact_property.property`
- `contactFields` state => `contactAttributes`
- `FieldsElement` => `AttribElement`
- `SetContactField` => `SetContactAttrib`
- `split_by_group` => `split_by_groups` (missed in previous PR)

@ericnewcomer thinking we should add a new term for contact fields now that goflow expects actions that update `fields` (custom contact properties) and `properties` (reserved contact properties), and 'attributes' seemed appropriate. LMK what you think, happy to use a different term. 

waiting for an initial review before adding tests 👍

also note that the deploy preview is 💩on account of the customer service flow definition change

### CODE REVIEW

**Comment by ericnewcomer** [comment]
In goflow it's `set_contact_property` and `set_contact_field`. Don't think attribute is a thing.

**Comment by rowanseymour** [comment]
We're soon gonna have these docs updated automatically with every release https://nyaruka.github.io/goflow/

**Comment by ycleptkellan** [comment]
sweet, those docs look great

@rowanseymour this PR creates `set_contact_property` actions in the format:

```
{
 "uuid": "64378fc1-19e4-4c8a-be27-aee49ebc728a",
 "value": "Some Person",
 "type": "set_contact_property",
 "property": "Name",
 "property_uuid": "a9827040-4773-43fd-b501-c1b2332eca42"
}
```

and `set_contact_property` actions in the format:

```
{
 "uuid": "64378fc1-19e4-4c8a-be27-aee49ebc728a",
 "value": "08/26/18",
 "type": "set_contact_field",
 "field": "Expected Delivery Date",
 "field_uuid": "23ff7152-b588-43e4-90de-fda77aeaf7c0"
}
```

are those expected to match their `Action` counterparts in the goflow?

@ericnewcomer right, i'm just classing `set_contact_property` and `set_contact_field` as 'attributes', is that unnecessary? right now we're favoring `set_contact_field` (it's name of the action component/form, `contactFields` is the name we have in state, etc.). LMK if i'm missing something

**Comment by rowanseymour** [comment]
@ycleptkellan what is `property_uuid`? There's no equivalent in goflow. Property is just one of "name" or "language" (or eventually "timezone")

**Comment by ycleptkellan** [comment]
@rowanseymour that's the uuid of the property itself while the `uuid` field is the uuid of the action

**Comment by ycleptkellan** [comment]
so do we want that format to be 

```
{
  "type": "set_contact_property",
  "uuid": "8eebd020-1af5-431c-b943-aa670fc74da9",
  "property": "language",
  "value": "eng"
}
```

where the `uuid` is the action uuid? or the property uuid?

**Comment by rowanseymour** [comment]
@ycleptkellan properties aren't like contactfields - they don't correspond to an asset with a UUID - they're just the named property of a contact (name, language, timezone). In your example `uuid` is the action UUID.

**Comment by ycleptkellan** [comment]
OK so we would want 

```
{
 "type": "set_contact_property",
 "uuid": "64378fc1-19e4-4c8a-be27-aee49ebc728a",
 "property": "name",
 "value": "Some Person",
}
```

using the data from my first example?

**Comment by ycleptkellan** [comment]
@ericnewcomer this one's ready for an initial (pre-test) review. fixes #221, addresses #215 and generally brings us to date w/ goflow's `set_contact_property`, `set_contact_field` actions. the most uncomfortable aspect of this PR might be the inclusion of 'Attribute'/'Attrib' as a prefix for components that read/create both `set_contact_property` and `set_contact_field` actions...

note that you'll need to pull it down as changes were made to both of our flow assets.

File: src/component/actions/SetContactAttrib/SetContactAttribForm.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    };
+
+    constructor(props: SetContactAttribFormProps, context: ConfigProviderContext) {
+        super(props);
+
+        this.onValid = this.onValid.bind(this);
+    }
+
+    public onValid(widgets: { [name: string]: any }): void {
+        const { wrappedInstance: { state: { value } } } = widgets.Value;
```
Comment (ericnewcomer) [inline src/component/actions/SetContactAttrib/SetContactAttribForm.tsx]: We really need to pull together a testing strategy for these

**Comment by ycleptkellan** [comment]
what happens if you click into `FormElement.scss` and then save?

**Pull request merged**